jobbergate-api/jobbergate_api/apps/job_scripts/routers.py

```diff
@@ -78,58 +77,47 @@ def render_template(template_files, param_dict_flat):
-        supporting_files = list()
-
-    default_template = [
-        default_template := param_dict["jobbergate_config"].get("default_template"),
-        "templates/" + default_template,
-    ]
-
-    template_files = {}
-    for member in s3_application_tar.getmembers():
-        if member.name in default_template:
-            contentfobj = s3_application_tar.extractfile(member)
-            template_files["application.sh"] = contentfobj.read().decode("utf-8")
-        if member.name in supporting_files:
-            match = [x for x in support_files_output if member.name in x]
-            contentfobj = s3_application_tar.extractfile(member)
-            filename = support_files_output[match[0]][0]
-            template_files[filename] = contentfobj.read().decode("utf-8")
-
-    # Use tempfile to generate .tar in memory - NOT write to disk
+    application_files = get_application_files_from_s3(application_id)
+
+    default_template_path = PurePath(application_config.jobbergate_config.default_template)
+    default_template_name = default_template_path.name
+
+    with UploadedFilesValidationError.check_expressions("Application files not found") as check:
+        check(
+            application_files.source_file,
+            f"Application file was not found for {application_id=}",
+        )
+        check(
+            application_files.templates,
+            f"No template was found for {application_id=}",
+        )
+        check(
+            default_template_name in application_files.templates,
+            f"{default_template_name=} was found for {application_id=}",
+        )
+
+    # rename default template
+    default_template_file = application_files.templates.pop(default_template_name)
+    application_files.templates["application.sh"] = default_template_file
+
     param_dict_flat = {}
-    for (key, value) in param_dict.items():
+    for (key, value) in application_config.dict().items():
         if isinstance(value, dict):
             for nest_key, nest_value in value.items():
                 param_dict_flat[nest_key] = nest_value
         else:
             param_dict_flat[key] = value
-    with tempfile.NamedTemporaryFile("wb", suffix=".tar.gz", delete=False) as f:
-        with tarfile.open(fileobj=f, mode="w:gz") as rendered_tar:
-            for member in s3_application_tar.getmembers():
-                if member.name in supporting_files:
-                    contentfobj = s3_application_tar.extractfile(member)
-                    supporting_file = contentfobj.read().decode("utf-8")
-                    template = Template(supporting_file)
-                    rendered_str = template.render(data=param_dict_flat)
-                    tarinfo = tarfile.TarInfo(member.name)
-                    rendered_tar.addfile(tarinfo, StringIO(rendered_str))
-        f.flush()
-        f.seek(0)
-
-    job_script_data_as_string = render_template(template_files, param_dict_flat)
+        print(key)
```

Comment:
left over

Reply:
solved in [5a74fef](https://github.com/omnivector-solutions/jobbergate/pull/128/commits/5a74fefad02eea5293a92f816e215a969c5cb1ea)